lib/shopify_cli/theme/extension/dev_server/watcher.rb

```diff
@@ -24,6 +28,18 @@ def start
           def stop
             @listener.stop
           end
+
+          def enqueue_updates(modified, added, _removed)
+            @syncer.enqueue_updates(filter_extension_files(modified + added))
+          end
+
+          private
+
+          def filter_extension_files(files)
+            files
+              .select { |f| @extension.extension_file?(f) }
```

Comment:
@karreiro for cases like this in the CLI, I feel like it would be really helpful to use `filter_map`, but it's only available in Ruby versions `>v.27`. CP mentioned on a thread that we may be moving to deprecate `v2.6.x` support – if so, we can maybe adopt this approach everywhere it's applicable?

Reply:
I believe we may consider the deprecation discussion is in progress until we update this field: https://github.com/Shopify/shopify-cli/blob/main/shopify-cli.gemspec#L18.

But, after we update that, +1 for the `filter_map`, it's much elegant indeed ☺️